string    name
bool      save_mix
float64[] curr_trans

---

bool  done